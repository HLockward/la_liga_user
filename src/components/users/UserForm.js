import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";

const UserForm = ({
  user,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{user.id ? "Edit" : "Add"} User</h2>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      <TextInput
        name="first_name"
        label="First Name"
        value={user.first_name}
        onChange={onChange}
        error={errors.first_name}
      />

      <TextInput
        name="last_name"
        label="Last Name"
        value={user.last_name}
        onChange={onChange}
        error={errors.last_name}
      />

      <TextInput
        name="email"
        label="Email"
        value={user.email}
        onChange={onChange}
        error={errors.email}
      />

      <TextInput
        name="avatar"
        label="Avatar"
        value={user.avatar}
        onChange={onChange}
        error={errors.avatar}
      />

      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
};

UserForm.propTypes = {
  user: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool
};

export default UserForm;
