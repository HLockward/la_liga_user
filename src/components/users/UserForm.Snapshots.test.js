import React from "react";
import UserForm from "./UserForm";
import renderer from "react-test-renderer";
import { users } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <UserForm
      user={users[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <UserForm
      user={users[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
