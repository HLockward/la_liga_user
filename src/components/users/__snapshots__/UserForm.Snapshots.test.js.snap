// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     User
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="first_name"
    >
      First Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="first_name"
        onChange={[MockFunction]}
        type="text"
        value="George"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="last_name"
    >
      Last Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="last_name"
        onChange={[MockFunction]}
        type="text"
        value="Bluth"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="email"
        onChange={[MockFunction]}
        type="text"
        value="george.bluth@reqres.in"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="avatar"
    >
      Avatar
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="avatar"
        onChange={[MockFunction]}
        type="text"
        value="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     User
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="first_name"
    >
      First Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="first_name"
        onChange={[MockFunction]}
        type="text"
        value="George"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="last_name"
    >
      Last Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="last_name"
        onChange={[MockFunction]}
        type="text"
        value="Bluth"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="email"
        onChange={[MockFunction]}
        type="text"
        value="george.bluth@reqres.in"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="avatar"
    >
      Avatar
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="avatar"
        onChange={[MockFunction]}
        type="text"
        value="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
